### DISCUSSION TITLE
installing gauge on s390x archetecture

### DISCUSSION THREAD

**Discussion by meshungina** [getgauge/gauge#2394] [General]
Can someone advise if it is possible to run on s390x? Any help is appreciated!

**Comment by chadlwilson** [comment]
I don't think the main project or any of the plugins for various languages (where they have golang involved) produce binaries for s390x.

Given the maintenance/contributor state of the project it might be difficult to take on more architectures and ensure they are validated. (Personal opinion)

Which plugin language are you interested in?

**Reply by meshungina** [reply]
The plugin languages are go and ruby. I'm trying to run some OpenShift Pipelines E2E tests which require gauge [](https://github.com/openshift-pipelines/release-tests/)

**Reply by chadlwilson** [reply]
Yeah none of [gauge itself](https://github.com/getgauge/gauge/releases/tag/v1.5.4), [gauge-go](https://github.com/getgauge-contrib/gauge-go/releases/tag/v0.4.0) or [gauge-ruby](https://github.com/getgauge/gauge-ruby/releases/tag/v0.7.0) have s390x binaries.

Only recently were there even gauge-ruby binaries for `linux/arm64` :-)

I guess you could try and change yourself to crosscompile for `linux/s390x` and see whether things generally work.

Just curious, why would you like the test specs themselves to run on s390x hardware? Rather than perhaps having tests that run on x64 or arm64 machines but drive the system-under-test externally that may themselves be running on s390x? Would it be possible to "drive" Openshift containers running on s390x nodes via Gauge runners which themselves are running separately on different x64/arm64 nodes?

**Reply by meshungina** [reply]
Good idea I will try driving the system-under-test from my local machine. I actually didn't consider this :) This is the repository that I'm looking at: https://github.com/openshift-pipelines/release-tests/ I am rather new to such things so appreciate already your help.

**Reply by chadlwilson** [reply]
Oh, wow, interesting - I haven't seen those, so not sure what they require or how they are designed to work (run internal to the cluster, or run externally from outside the cluster). But worth trying that first.
